File: 03-sliding-window/01-minimum-size-subarray-sum/Solution.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        l = 0
        total = 0
        best = float('inf')
        for r in range(len(nums)):
            total += nums[r]
            while total >= target:
                best = min(best, r - l + 1)
                total -= nums[l]
                l += 1
        return best if best != float('inf') else 0

File: 03-sliding-window/01-minimum-size-subarray-sum/Solution.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l = 0, total = 0, best = Integer.MAX_VALUE;
        for (int r = 0; r < nums.length; r++) {
            total += nums[r];
            while (total >= target) {
                best = Math.min(best, r - l + 1);
                total -= nums[l];
                l++;
            }
        }
        return best == Integer.MAX_VALUE ? 0 : best;
    }
}
